logger/logger_writer.go

```diff
@@ -146,6 +147,8 @@ func queueMsg(lvl Level, prefix, format string, v ...interface{}) (err error) {
 		return
 	}
 
+	msg.time = time.Now()
+
```

Comment:
I think you're right about the msg.Reset, Dham.  Except I think I'd actually put it inside freeMsg (and get rid of the pre-freeMsg call at line 225) -- you really never want to put a message back on the free channel without resetting it, right?

The defer inside the select looks a little funny, but since the only other case in the select ends in a return, I don't think there's any difference between deferring it there vs immediately after the select.  Is there?

In any case, I think the error path in this func is essentially never exercised -- bytes.Buffer.Write() seems to never return an error, only panic if the buffer can no longer grow.